h2/src/main/org/h2/command/dml/Optimizer.java

```diff
@@ -100,9 +100,7 @@ private boolean canStop(int x) {
         if ((x & 127) == 0) {
             long t = System.nanoTime() - startNs;
             // don't calculate for simple queries (no rows or so)
-            if (cost >= 0 && 10 * t > cost * TimeUnit.MILLISECONDS.toNanos(1)) {
-                return true;
-            }
+            return cost >= 0 && 10 * t > cost * TimeUnit.MILLISECONDS.toNanos(1);
```

Comment:
This change actually generates worse and larger bytecode than it was before.

Reply:
that's interesting too. How about current version?

Reply:
If was expected, result of comparison is not a real `boolean` value in JVM so additional instructions are required.

```Java
a = <comparison>
```
is compiled like
```Java
if (<comparison>)
    a = true;
else
    a = false;
```
Current version saves 2 bytes but now this current is less readable.

Reply:
I missed your question about condition.

`x & 127` is here to avoid execution of slower inner condition on each invocation. This is more or less oblivious.